src/can.rs

```diff
@@ -1,657 +1,94 @@
-    Fifo1Full,
-    /// FIFO 0 overrun interrupt enable
-    Fifo0Ovr,
-    /// FIFO 1 overrun interrupt enable
-    Fifo1Ovr,
-    /// Transmit mailbox empty interrupt
-    Txe,
-}
-
-impl embedded_hal_can::Id for CanId {
-    type BaseId = u16;
-    type ExtendedId = u32;
-
-    fn base_id(&self) -> Option<<CanId as Id>::BaseId> {
-        match self {
-            CanId::BaseId(v) => Some(*v),
-            CanId::ExtendedId(_) => None,
-        }
-    }
-
-    fn extended_id(&self) -> Option<<CanId as Id>::ExtendedId> {
-        match self {
-            CanId::BaseId(_) => None,
-            CanId::ExtendedId(v) => Some(*v),
-        }
+        impl sealed::Sealed for gpiod::PD0<AF7<PushPull>> {}
+        impl RxPin for gpiod::PD0<AF7<PushPull>> {}
+        impl sealed::Sealed for gpiod::PD1<AF7<PushPull>> {}
+        impl TxPin for gpiod::PD1<AF7<PushPull>> {}
     }
 }
 
-impl embedded_hal_can::Frame for CanFrame {
-    type Id = CanId;
-
-    #[inline(always)]
-    fn is_remote_frame(&self) -> bool {
-        self.data().is_none()
-    }
-
-    #[inline(always)]
-    fn is_data_frame(&self) -> bool {
-        self.data().is_some()
-    }
-
-    fn id(&self) -> <CanFrame as Frame>::Id {
-        self.id
-    }
-
-    #[inline(always)]
-    fn data(&self) -> Option<&[u8]> {
-        if self.dlc > 0 {
-            Some(&self.data)
-        } else {
-            None
-        }
-    }
-}
-
-impl embedded_hal_can::Filter for CanFilter {
-    type Id = CanId;
-
-    /// Construct a filter which filters messages for a specific identified
-    fn from_id(id: Self::Id) -> Self {
-        CanFilter::new(CanFilterData::IdFilter(id))
-    }
-
-    /// Construct an "empty" filter which will accept all messages
-    fn accept_all() -> Self {
-        CanFilter::new(CanFilterData::AcceptAll)
-    }
-
-    // TODO: Constructing filters like this is fairly limiting because ideally
-    // we would have the full "filter state" available, so for non-extended
-    // filters this could be 2 masks and filters or 4 ids for id lists
-
-    /// Constuct a mask filter. This method accepts two parameters, the mask which designates which
-    /// bits are actually matched againts and the filter, with the actual bits to match.
-    fn from_mask(mask: u32, filter: u32) -> Self {
-        crate::assert!(
-            mask < MAX_EXTENDED_ID,
-            "Mask cannot have bits higher than 29"
-        );
-
-        // If the mask has any significant (i.e. 1) bits in the upper bits, this filter is extended
-        if (mask & EXID_MASK) > 0 {
-            CanFilter::new(CanFilterData::ExtendedMaskFilter(mask, filter))
-        } else {
-            CanFilter::new(CanFilterData::MaskFilter(mask as u16, filter as u16))
-        }
-    }
+/// Struct representing a CAN peripheral and its configured TX and RX pins.
+///
+/// See [`bxcan::Instance`] for more information on how to use the CAN interface.
+pub struct Can<TX, RX> {
+    _can: pac::CAN,
+    _tx: TX,
+    _rx: RX,
```

Comment:
When the `free()` method is implemented, the underscore should'nt be necessary anymore:

```suggestion
pub struct Can<TX, RX> {
    can: pac::CAN,
    tx: TX,
    rx: RX,
```